### ISSUE TITLE
[2011.09471] FROST: Faster and more Robust One-shot Semi-supervised Training

### ISSUE THREAD

**Issue by riron1206** [riron1206/read_paper#33]
## 概要
https://arxiv.org/abs/2011.09471
FROST
one-shot semi-supervised learning （各クラス1サンプルのみの分類を半教師）の手法
FixNatchより早い、高精度、ロバスト
Imbalanced unlabeled にしたCIFAR-10でも精度でており
ワンショット半教師ありトレーニングの最も実用的な方法とのこと

- FixMatch + BootStrapping = FROST
- BOSS + BootStrapping = FROST+p+b

![fig](https://github.com/HelenaELiu/FROST/raw/master/TF-FROST/images/teaser.png?raw=true)

## 新規性・差分
- 学習中に確信度が高いサンプルの数を増やすBootStrapping（この論文の一番の新規性）
- 教師なしの2種のlossを使う
  Augmentationの強弱で予測不変(Consistency Regularization)と表現一致(Contrastive Learning)を調整
  loss = 教師ありのloss + Consistency Regularizationの教師なしのloss + Contrastive Learningの教師なしのloss

※Consistency Regularization: FixMatchで使われてる元の画像に別の変換をかけたものの予測値の差分を目的関数に組み込むことによって、ラベルなしデータの仮ラベルの予測値に信頼性を持たせる
※Contrastive Learning: SimCLRで使われてる同じクラスのサンプルは近づけ違うクラスのサンプルは遠ざける。似ているデータは潜在空間でも似た埋め込みベクトルになり、異なるデータは潜在空間でも異なる埋め込みベクトルになる

## 結果
CIFAR-100, SVHN, and STL-10でFixMatch、BOSS よりずっと高精度

## コメント
- 公式のgithub: 
    - TF, pytorch版どちらもあり
    - https://github.com/HelenaELiu/FROST

<br>

- 参考サイト
    - https://www.getrevue.co/profile/icoxfog417/issues/weekly-machine-learning-204-295243

## その他メモ
■one-shot semi-supervised learning （各クラス1サンプルのみの分類を半教師）の強い手法
1. BOSS
    - https://arxiv.org/abs/2006.09363
2. FixMatch 

■他の半教師付き学習（各クラスあたり25サンプル以下の半教師）
1. π model
2. temporal ensembling
3. VAT
4. ICT
5. UDA
6. S4L
7. MixMatch
8. Mean Teachers

■クラス・バランシング手法（各クラスのサンプル数の不均衡を補正する手法）
1. データレベルの手法
    - 少数のクラスをオーバーサンプルし、多数のクラスをアンダーサンプル
2. アルゴリズムレベルの手法
    - 少数派クラスに属する学習サンプルには損失関数の中でより大きな重みを使用し，多数派クラスのサンプルにはより小さな重みを使用
